### ISSUE TITLE
Implement an Advanced configuration editor for NVDA

### ISSUE THREAD

**Issue by derekriemer** [nvaccess/nvda#6788]
I was thinking the other day about how to access settings that aren't accessible from the GUI in a slightly safer way. I came up with a scheme similar to the mozilla about:config editor.
http://www.instantbird.com/

* call the advancedConfig function in the python console of NVDA.

A dialog appears. There's a search field, a list of config keys and their values.
Keys are placed in the dialog by concatenating every item in the key with a ".". For example, "general.language", "virtualBuffers.passThroughAudioIndication", "speech.speechPlayerInEspeak.rateBoost", "schemaVersion"

* When the list appears, either find the configuration value you'd like, or press shift+tab to filter.
* If you are in the search box, type anything. Space separated words are searched separately, and items that match all given tokens are considered. Case is ignored.
* Example search: "braille cursor blink" shows  
    ```
    Key Value
    braille.cursorBlinkRate 500
    braille.cursorBlink True
    ```  
    "braille blink cursor" yields the same results.  
    "debugLog" shows all items with debugLog somewhere in the key.  
    "debugLog." shows all items with a section called debugLog.  
    .log shows only items that are below some section with log in them.
* search isn't currently powerful enough to search only in the last node of the dotted key.
* search can't currently ask for only items that match the first node of the dotted key.

Actions available from the list:

* When enter is pressed on a key in  the list, a dialog appears with either a checkbox, or edit field to edit said config value.
    * If okay is pressed in this dialog, the config and UI are updated.
    * if cancel is pressed, Well, it should be obvious, nothing happens.
* if delete is pressed on a key in the list, that key is deleted if the user says yes.
* Currently, okay/cancel do nothing at all in the main dialog.

Thoughts on other things to implement or concerns before I create a pull request?

**Comment by leonardder** [comment]
Sounds interesting. If implemented, it would be good if this editor would be able to remove unused config keys from configuration profiles.
I see no reason to hide this in the python console, just add an advanced config editor to the tools menu.

N.B. Sent on behalf of @BabbageCom

**Comment by derekriemer** [comment]
> Sounds interesting. If implemented, it would be good if this editor would be able to remove unused config keys from configuration profiles.

I'd like to do this. Deleting keys is insanely hard, I have no idea how to reliably do it outside of what I have.

> I see no reason to hide this in the python console, just add an advanced config editor to the tools menu.

My concern is that really, really bad things could happen if editing config wrongly. I.E. changing the no braille port to "Bluetooth", or disabling UIA.

>

**Comment by josephsl** [comment]
Hi, I’d say it should be something power users and developers should use, thus I’d vote for this be invoked by typing the proposed string from the Python Console. Thanks.

**Comment by kaveinthran** [comment]
agree, as this is a power user thing, it should be typed, and this way
is the standard used by mozilla

**Comment by derekriemer** [comment]
Actually, in firefox, you can get advanced settings, but it's way way berried in the GUI.

**Comment by derekriemer** [comment]
This now does a spin control for ints

**Comment by Adriani90** [comment]
This would be a really nice feature indeed.
cc: @feerrenrut, @michaelDCurran
